client/utils/date-time.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { dateI18n } from '@wordpress/date';
+
+type DateTimeFormat = string | null;
+
+interface FormatDateTimeOptions {
+	includeTime?: boolean; // Whether to include time in the formatted string (defaults to true)
+	useGmt?: boolean; // Whether to display the time in GMT/UTC (defaults to false)
+	separator?: string; // Separator between date and time (defaults to ' / ')
+	customFormat?: DateTimeFormat; // Custom format to use instead of WordPress settings
+}
+
+/**
+ * Formats a date and time string according to WordPress settings or a custom format.
+ *
+ * @param { string | Date } dateTime - The date and time string (e.g., '2024-10-23 15:28:26') or a JS Date object.
+ * @param { FormatDateTimeOptions } options - Additional options to control time inclusion and whether to use GMT/UTC.
+ * @return { string } - The formatted date and time string.
+ */
```

Comment:
It may help future devs not make mistakes here by clarifying that the `formatUserDateTime` expects `dateTime: string | Date` to be passed in as a UTC date or string, which is then converted to the WP Settings TZ. Otherwise, `datei18n` will parse the string as the browser's local timezone.

```suggestion
/**
 * Formats a date and time string according to WordPress settings or a custom format.
 *
 * @param { string | Date } dateTime - The date and time string (e.g. '2024-10-23 15:28:26') or a JS Date object. `dateTime` strings without a timezone suffix (e.g. `Z` for UTC) will be interpreted as the browser's local timezone.
 * @param { FormatDateTimeOptions } options - Additional options to control time inclusion and whether to use GMT/UTC.
 * @return { string } - The formatted date and time string.
 */
```

Reply:
I'd say this comment is no longer relevant after last changes. But I made sure to mention that the input dates are converted to UTC for consistency e.g. [here](https://github.com/Automattic/woocommerce-payments/blob/933422e00040f48a13487650adc0c1615f8aa8af/client/utils/date-time.ts#L22) and here https://github.com/Automattic/woocommerce-payments/blob/933422e00040f48a13487650adc0c1615f8aa8af/client/utils/date-time.ts#L54